app/models/manageiq/providers/openstack/inventory/persister/definitions/cloud_collections.rb

```diff
@@ -123,9 +123,10 @@ def add_key_pairs(extra_properties = {})
     add_collection(cloud, :key_pairs, extra_properties) do |builder|
       builder.add_properties(
         :model_class => ManageIQ::Providers::Openstack::CloudManager::AuthKeyPair,
+        # targeted refresh workaround-- always refresh the whole keypair collection
+        :targeted    => false
```

Comment:
I think we still need to conditionalize this , it should copy what we do in https://github.com/ManageIQ/manageiq-providers-openstack/pull/323/files#diff-0698b10bec7e7c31e8eabb30dc507213R140

So we fetch all entities only if there is something in references. 
---

Now, it the references would be blank, we would delete all keypairs